DigitalLearningSolutions.Data/Services/CourseContentService.cs

```diff
@@ -23,21 +24,53 @@ public CourseContentService(IDbConnection connection, ILogger<CourseContentServi
             this.logger = logger;
         }
 
-        public CourseContent? GetCourseContent(int customisationId)
+        public CourseContent? GetCourseContent(int candidateId, int customisationId)
         {
-            return connection.QueryFirstOrDefault<CourseContent>(
+            CourseContent? courseContent = null;
+            return connection.Query<CourseContent, CourseSection, CourseContent>(
                 @"SELECT Customisations.CustomisationID AS id,
                          Applications.ApplicationName,
-	                     Customisations.CustomisationName,
-	                     dbo.GetMinsForCustomisation(Customisations.CustomisationID) AS AverageDuration,
-	                     Centres.CentreName,
-	                     Centres.BannerText
+                         Customisations.CustomisationName,
+                         dbo.GetMinsForCustomisation(Customisations.CustomisationID) AS AverageDuration,
+                         Centres.CentreName,
+                         Centres.BannerText,
+                         Sections.SectionName,
+                         dbo.CheckCustomisationSectionHasLearning(Customisations.CustomisationID, Sections.SectionID) AS HasLearning,
+                         (CASE
+                            WHEN Progress.CandidateID IS NULL
+                                 OR dbo.CheckCustomisationSectionHasLearning(Customisations.CustomisationID, Sections.SectionID) = 0
+                            THEN 0
+                            ELSE CAST(SUM(aspProgress.TutStat) * 100 AS FLOAT) / (COUNT(Tutorials.TutorialID) * 2)
+                         END) AS PercentComplete
                   FROM Applications
                   INNER JOIN Customisations ON Applications.ApplicationID = Customisations.ApplicationID
+                  INNER JOIN Sections ON Sections.ApplicationID = Applications.ApplicationID
                   INNER JOIN Centres ON Customisations.CentreID = Centres.CentreID
-                  WHERE Customisations.CustomisationID = @customisationId;",
-                new { customisationId }
-            );
+                  LEFT JOIN Tutorials ON Sections.SectionID = Tutorials.SectionID
+                  LEFT JOIN Progress ON Customisations.CustomisationID = Progress.CustomisationID AND Progress.CandidateID = @candidateId

```

Comment:
You also need to check here that SystemRefreshed is 0 and RemovedDate is NULL 🙂

Reply:
Sorry I hadn't realised that those fields needed to be checked! This query was closely based on the `uspReturnSectionsForCandCust_V2` stored procedure and I don't think this was being checked there. Then again I don't understand everything going on in that query, so it might have been being checked in a different way.

I've added a check in 48d2aed 🙂

Reply:
I assume the stored procedure doesn't check that because it's checking for a specific progress id 🙂 And yes I should have mentioned those fields needed checking earlier, I only recently realised when helping Frank with his ticket. When the old code fetches a progress id it will check that the customisation id and candidate id match + checking the removed date and system refreshed. That's because there can be more than one progress entry per user & customisation. For example if they unenrol from a course, a new progress entry should be created if they enrol again 🙂

Reply:
Ah I see! Thanks for explaining that 🙂